pyproject.toml

```diff
@@ -1,7 +1,83 @@
 [build-system]
-requires = ["setuptools>=46.4.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = [
+    "hatchling>=1.3.1",
+]
+build-backend = "hatchling.build"
```

Comment:
hmm, would rather ebp just standardised around flit, which is already now used by most packages, unless there is a particular reason not to

Reply:
Sure, I prefer hatchling but I hadn't appreciated that we're a mainly flit workshop at this stage.

Reply:
@chrisjsewell can you move this to #98

Reply:
> unless there is a particular reason

Hey, Hatch maintainer here! One reason is for plugins, so you could do things like versioning using just Git or compiling your codebase with Mypyc to speed things up.

Also, Project Jupyter is almost done [switching](https://github.com/jupyterlab/team-compass/issues/146) their entire packaging ecosystem. I tried to link to most of the changes here: https://hatch.pypa.io/latest/users/

Reply:
Yes, personally I'm a big fan of Hatch over Flit. I think Flit is a great tool, but Hatch is more customisable / extensible in my experience.

The nice thing is that with PEP 631 (thanks @ofek) we can switch fairly easily between the two.